### PR TITLE
Run3-hcx327 Use of ESGetToken in the SiwWatcher codes in Validation/HcalHits

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Use of ESGetToken in the SiwWatcher codes in Validation/HcalHits

#### PR validation:

Use test workflows in Validation/HcalHits/test

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by jfernan2** [comment]
+1

**Comment by qliphy** [comment]
+1

File: Validation/HcalHits/src/SimG4HcalValidation.cc [hunk 1]
```diff
@@ -223,13 +229,10 @@ void SimG4HcalValidation::init() {
...
+void SimG4HcalValidation::beginRun(edm::EventSetup const &es) {
   // Numbering From DDD
-  edm::ESHandle<HcalDDDSimConstants> hdc;
-  (*job)()->get<HcalSimNumberingRecord>().get(hdc);
-  const HcalDDDSimConstants *hcons = hdc.product();
-  edm::LogVerbatim("ValidHcal") << "HcalTestAnalysis:: Initialise "
-                                << "HcalNumberingFromDDD";
+  const HcalDDDSimConstants *hcons = &es.getData(ddconsToken_);
+  edm::LogVerbatim("ValidHcal") << "HcalTestAnalysis:: Initialise HcalNumberingFromDDD";
   numberingFromDDD = new HcalNumberingFromDDD(hcons);
```
Comment (perrotta) [inline Validation/HcalHits/src/SimG4HcalValidation.cc:236]: Just checking...
Previously the code made `new` `numberingFromDDD` and `org` at the begin of the job, and then `numberingFromDDD` was deleted in the destructor.
Now `numberingFromDDD` and `org` are `new`ed at every `beginRun`, but still deleted only once.
Is this the wanted behaviour?

Comment (bsunanda) [reply Validation/HcalHits/src/SimG4HcalValidation.cc:236]: I shall take care of this once I get the IB with the merged PR. It is always not straight forward to take care of an old piece of code

**Pull request merged**